GooglePlayInstant/Editor/AppBundleBuilder.cs

```diff
@@ -0,0 +1,202 @@
+    /// </summary>
+    public static class AppBundleBuilder
+    {
+        private const string BaseModuleZipFileName = "base.zip";
+
+        public static void Build(string aabFilePath)
+        {
+            var tempFilePath = Path.GetTempFileName();
+            Debug.LogFormat("Building Package: {0}", tempFilePath);
+            var buildPlayerOptions = PlayInstantBuilder.CreateBuildPlayerOptions(tempFilePath, BuildOptions.None);
+
+            // Do not use BuildAndSign since this signature won't be used.
+            if (!PlayInstantBuilder.Build(buildPlayerOptions))
+            {
+                // Do not log here. The method we called was responsible for logging.
+                return;
+            }
+
+            CreateAppBundle(tempFilePath, aabFilePath);
+        }
+
+        // TODO: currently all processing is synchronous; consider moving to a separate thread
+        private static void CreateAppBundle(string inputFile, string aabFilePath)
+        {
+            try
+            {
+                DisplayProgress("Running aapt2", 0.2f);
+                var workingDirectory = new DirectoryInfo(Path.Combine(Path.GetTempPath(), "play-instant-unity"));
+                if (workingDirectory.Exists)
+                {
+                    workingDirectory.Delete(true);
+                }
+
+                workingDirectory.Create();
+                var sourceDirectoryInfo = workingDirectory.CreateSubdirectory("source");
+                var destinationDirectoryInfo = workingDirectory.CreateSubdirectory("destination");
+
+                var tempFileName = Path.GetRandomFileName();
+                var tempFilePath = Path.Combine(sourceDirectoryInfo.FullName, tempFileName);
+                var aaptResult = AndroidAssetPackagingTool.Convert(inputFile, tempFilePath);
+                if (aaptResult != null)
+                {
+                    LogError("aapt2", aaptResult);
+                    return;
+                }
+
+                DisplayProgress("Creating base module", 0.4f);
+                var unzipFileResult = ZipUtils.UnzipFile(tempFileName, sourceDirectoryInfo.FullName);
+                if (unzipFileResult != null)
+                {
+                    LogError("Unzip", unzipFileResult);
+                    return;
+                }
+
+                File.Delete(tempFilePath);
+
+                var baseModuleZip = Path.Combine(workingDirectory.FullName, BaseModuleZipFileName);
+                ConvertFiles(sourceDirectoryInfo, destinationDirectoryInfo);
+
+                var zipFileResult = ZipUtils.CreateZipFile(baseModuleZip, destinationDirectoryInfo.FullName, ".");
+                if (zipFileResult != null)
+                {
+                    LogError("Zip creation", zipFileResult);
+                    return;
+                }
+
+                DisplayProgress("Running bundletool", 0.6f);
+                var buildBundleResult = Bundletool.BuildBundle(baseModuleZip, aabFilePath);
```

Comment:
This call breaks with the below error if you try to overwrite a previously built app bundle. We should either delete the app bundle if it already exists or build to a temp file and overwrite the old once the build finishes.

```
bundletool failed: 

Failed to run '/Library/Java/JavaVirtualMachines/jdk1.8.0_181.jdk/Contents/Home/bin/java -jar "/Users/jakecohen/Projects/Unity/GPITestProject/Library/bundletool-all-0.6.0.jar" build-bundle --config="/var/folders/g9/y77dt2r96p1g700j3pvx83_c00k690/T/BundleConfig.json" --modules="/var/folders/g9/y77dt2r96p1g700j3pvx83_c00k690/T/play-instant-unity/base.zip" --output="/Users/jakecohen/Projects/Unity/build/bundle.aab"'
stdout:
[BT:0.6.0] Error: File '/Users/jakecohen/Projects/Unity/build/bundle.aab' already exists.
java.lang.IllegalArgumentException: File '/Users/jakecohen/Projects/Unity/build/bundle.aab' already exists.
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:204)
	at com.android.tools.build.bundletool.utils.files.FilePreconditions.checkFileDoesNotExist(FilePreconditions.java:32)
	at com.android.tools.build.bundletool.commands.BuildBundleCommand.validateInput(BuildBundleCommand.java:242)
	at com.android.tools.build.bundletool.commands.BuildBundleCommand.execute(BuildBundleCommand.java:162)
	at com.android.tools.build.bundletool.BundleToolMain.main(BundleToolMain.java:69)
	at com.android.tools.build.bundletool.BundleToolMain.main(BundleToolMain.java:44)

stderr:

exit code: 1
```

Reply:
Good catch! Fixed by deleting the existing file. I confirmed that `EditorUtility.SaveFilePanel()` already prompts for whether to replace existing, so deleting seems fair.